fe1-web/src/features/lao/reducer/ServerReducer.ts

```diff
@@ -0,0 +1,197 @@
+/**
+ * This error is disabled since reducers use the createSlice function, which requires the user to
+ * param-reassign. Please do not disable other errors.
+ */
+/* eslint-disable no-param-reassign */
+import { createSelector, createSlice, Draft, PayloadAction } from '@reduxjs/toolkit';
+
+import { PublicKey } from 'core/objects';
+
+import { Server, ServerAddress, ServerState } from '../objects/Server';
+import { getLaosState } from './LaoReducer';
+
+/**
+ * Reducer & associated function implementation to store all known servers
+ */
+
+export interface ServerReducerState {
+  byLaoId: {
+    [laoId: string]: {
+      byAddress: {
+        [address: ServerAddress]: ServerState;
+      };
+      allAddresses: ServerAddress[];
+    };
+  };
+  backendKeyByFrontendKey: {
+    [frontendKey: string]: string;
+  };
```

Comment:
shouldn't this be contained within the `byLaoId` ?

Organizers front-ends might want to use different public keys in different LAOs (e.g., you don't necessarily want to reveal that you're both the organizer for `PoP students` LAO and `Let's kill the TA` LAO), but they might also use the same front-end keys with different backends (e.g., current implementation but with different servers for different student associations).
In the former case, your system would work, while it would break in the latter.

Reply:
Oops, yes that is problematic!